Mathlib/Topology/ContinuousOn.lean

```diff
@@ -1498,6 +1498,13 @@ lemma ContinuousOn.union_continuousAt
   (fun h => ContinuousWithinAt.continuousAt (continuousWithinAt hs h) <| IsOpen.mem_nhds s_op h)
   (ht _)
 
+/-- If a function is continuous on the union of two sets, it is continuous on both of them. -/
+theorem continuousOn_of_union {X Y : Type*} [TopologicalSpace X] [TopologicalSpace Y]
+    {s t : Set X} {f : X → Y} (h : ContinuousOn f (s ∪ t)) :
+    ContinuousOn f s ∧ ContinuousOn f t :=
```

Comment:
It's a bit unusual to have a theorem whose conclusion is a conjunction of statements. Usually, one would have separate statements for each.